doc/src/bond_cpp.md

```diff
@@ -185,6 +185,59 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Type evolution
+==============
+
+Bond does not use explicit versioning to deal with changes to schemas over time
+Instead, Bond supports certain type evolution operations which allow the
+producer and consumer of Bond types to evolve without lockstep coordination.
+
+The following changes to a type will never break compatibility across the wire:
+
+- Adding or removing an `optional` or `optional_required` field
+- Changing type between `int32` and `enum`
+- Changing type between `vector<T>` and `list<T>`
+- Changing type between `blob` and `vector<uint8>` or `blob` and `list<uint8>`
+- Changing type between `T` and `bonded<T>`
+- Adding new enumeration values at the end of an enumeration
+
+The following changes to a type are generally safe but require some
+consideration about how the change is rolled out:
+
+- Changing between `optional` and `required_optional` or between
+`required_optional` and `required`. This facilitates a two-step process for
+changing between `optional` and `required` but the first step must be completed
+on both the producer and the consumer sides before the second step can be
+started on either side.
+- Promoting a numeric type from a smaller size to a larger size (e.g.: `float`
+to `double`, `uint8` to `uint16`, `uint8` to `uint32`, `int8` to `int16`,
+etc.). The consumer must get the change before the producer.
+- Promoting from `int8` or `int16` to `enum`. The consumer must get the change
+before the producer.
+
+These following changes will break wire compatibility and are not recommended:
+
+- Adding or removing `required` fields
+- Incompatible change of field types (any type change NOT covered above): e.g.:
+`int32` to `string`, `string` to `wstring`
+- Changing of field ordinals
+- Changing of inheritance hierarchy (add/remove/change base struct)
```

Comment:
I would just say "add/remove base struct". You can change the fields inside a base struct if you follow the rules above.